cmake/target_toolchain.cmake

```diff
@@ -0,0 +1,31 @@
+set(CMAKE_SYSTEM_NAME      Generic)
+set(CMAKE_SYSTEM_PROCESSOR ${ARCH})
+
+set(BUILD_SHARED_LIBS OFF)
+
+set(CMAKE_FIND_ROOT_PATH_MODE_PROGRAM NEVER)
+set(CMAKE_FIND_ROOT_PATH_MODE_LIBRARY ONLY)
+set(CMAKE_FIND_ROOT_PATH_MODE_INCLUDE ONLY)
+set(CMAKE_FIND_ROOT_PATH_MODE_PACKAGE ONLY)
+
+if(NOT (COMPILER STREQUAL "host-gcc"))
+  include(${TOOLCHAIN_ROOT}/cmake/toolchain/${ZEPHYR_TOOLCHAIN_VARIANT}/target.cmake)
+endif()
+
+# Configure the toolchain based on what toolchain technology is used
+# (gcc, host-gcc etc.)
+include(${ZEPHYR_BASE}/cmake/compiler/${COMPILER}/target.cmake OPTIONAL)
```

Comment:
Dissonance between base directories for compiler and toolchain folders:
Above for host-gcc, `TOOLCHAIN_ROOT` was used, but here we use `ZEPHYR_BASE`.
Should be `TOOLCHAIN_ROOT` in both places as that works with out-of-tree stuff.

Reply:
(`TOOLCHAIN_ROOT` is btw ill-named as it is not the root dir of any toolchain -- just the dir holding some cmake files for toolchains.)

Reply:
What if you want to combine a compiler from Zephyr with an out-of-tree toolchain?

I suspect this code is working-as-intended.

Reply:
Even if just for consistency, they should be the same.
I believe Oticon is currently the main (only?) user of out-of-tree due to commercial custom toolchain. 

Anyway, I can fix this later.